src/DataNodeInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+namespace Dhii\Data\Tree;
+
+use Dhii\Data\Container\ContainerInterface;
+
+/**
+ * A node that can represent arbitrary data, and can be a part of a child-aware node tree.
+ * 
+ * Although the node's value and key SHOULD now be retrievable via the generic
```

Comment:
Consider "MAY be retrieved" instead of "SHOULD now be retrievable".

My reasoning is that a data container is an object that can provide a value that corresponds to a key. It is nowhere bound to have to provide values about itself; the attribution and ownership of the retrievable data is unrestricted.

Using the term SHOULD implies that implementations are considered ideal if they can provide the key and value through `get()` while implementations that don't are considered black sheep (_since they do not do what they SHOULD do_).

Using the term MAY leaves this behavior completely open and allows implementations to only provide the key and value if it fits the purpose.

Perhaps it may also avoid implementations having to inject the key and value into the data set or intercept calls to `get()` and simulate the existence of the key and value in the data set.